chromium_src/content/browser/tld_ephemeral_storage.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/chromium_src/content/public/browser/tld_ephemeral_storage.h"
+
+#include <map>
+#include "base/no_destructor.h"
+#include "content/browser/dom_storage/dom_storage_context_wrapper.h"
```

Comment:
there are many unused headers, probably if we drop them we could move the whole class outside of `chromium_src`?

Reply:
I will definitely remove the unused headers, but the idea is that eventually the private API exposed in browser_context.h can move into this class and it can eventually handle keeping track of all TLD-associated ephemeral DOM storage as well.